snuba/datasets/cdc/cdcprocessors.py

```diff
@@ -1,13 +1,30 @@
 from __future__ import annotations
+import re
 
 from abc import ABC, abstractmethod
+from datetime import datetime
 from typing import Any, Mapping, Optional, Sequence, Type
 
 from snuba.processor import MessageProcessor
 from snuba.writer import WriterTableRow
 
 KAFKA_ONLY_PARTITION = 0  # CDC only works with single partition topics. So partition must be 0
 
+POSTGRES_DATE_FORMAT_WITH_NS = "%Y-%m-%d %H:%M:%S.%f%z"
+POSTGRES_DATE_FORMAT_WITHOUT_NS = "%Y-%m-%d %H:%M:%S%z"
+
+date_with_nanosec = re.compile("^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.")
+
+
+def parse_postgres_datetime(date: str) -> datetime:
+    # Postgres dates express the timezone in hours while strptime expects
+    # the timezone to be expressed in HHmm, thus we need to add 00 as minutes.
+    date = f"{date}00"
+    if (date_with_nanosec.match(date)):
```

Comment:
Couldn't this be faster to check with just a `len(date)`? Without nanonseconds shoul dbe 20 characters long exactly always, right? Then not sure how much precision comes from nanoseconds, but you should be able to do a faster check with len.

Reply:
It would be faster indeed. I added a regex for consistency (to ensure the format was actually right), and did not improve it further because, according to cProfile, the impact of the regex match was negligible compared to the other improvements